irohad/util/utility_service.cpp

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "util/utility_service.hpp"
+
+#include <map>
+#include <set>
+
+#include "util/proto_status_tools.hpp"
+
+#include <boost/optional.hpp>
+#include <rxcpp/rx.hpp>
+#include "common/run_loop_handler.hpp"
+#include "logger/logger.hpp"
+#include "util/status.hpp"
+
+using namespace iroha::utility_service;
+
+static bool isFinalStatus(const Status &status) {
+  static const std::set<Status> final_statuses(
+      {Status::kStopped, Status::kFailed});
+  return final_statuses.count(status);
+}
```

Comment:
return (status == Status::kStopped) || (status == Status::kFailed);
All the same, you need to watch the method if you will correct it.